### PR TITLE
Decouple contract.

### PR DESCRIPTION
**Pull request by saipramod**

### CODE REVIEW

File: src/projection/projection.tsx [hunk 1]
```diff
@@ -2,8 +2,8 @@ import * as React from "react";
 import * as Highcharts from "highcharts";
 import more from "highcharts/highcharts-more";
 import HighchartsReact from "highcharts-react-official";
-import { getRangeData } from "../service";
-import { AreaRangeData } from "./type";
+import { getTimeSeriesData } from "../service";
```
Comment (lzl124631x) [inline src/projection/projection.tsx:5]: TimeSeriesData is too general. I might name it ProjectionData. (Not better to a great extent but at least closer to meaning of this app)

Comment (lzl124631x) [reply src/projection/projection.tsx:5]: I'm regarding the component I created as a Map visualization while yours is a Projection component

File: src/projection/projection.tsx [hunk 2]
```diff
@@ -19,15 +19,14 @@ interface AreaRangeState {
...
+    timeSeriesData: number[],
     maxValue: number,
     type: string,
     index: number
 ): Highcharts.Options => {
-    const title =
-        rangeData.contact === "100"
-            ? "No intervention"
-            : rangeData.contact + "% contact";
+    const contact = contactData.contact.replace("data", "");
```
Comment (lzl124631x) [inline src/projection/projection.tsx:28]: we shouldn't have a contact with value `data100`. Let's just return `100`?

File: src/projection/util.ts [hunk 3]
```diff
@@ -13,47 +42,54 @@ export function toAreaRangeSeries(data: AreaRangeData, index: number): any[] {
...
+    )?.data;
+    const percentiledata25 = data.percentileData.find(
+        (_) => _.percentile == "25"
+    )?.data;
+    const percentiledata50 = data.percentileData.find(
+        (_) => _.percentile == "50"
+    )?.data;
+    const percentiledata75 = data.percentileData.find(
+        (_) => _.percentile == "75"
+    )?.data;
```
Comment (lzl124631x) [inline src/projection/util.ts:57]: this is a bit too hard coded. I'd sort the percentileData in ascending order of percentile, and in front-end I merge 1 and N-1 as arearange, 2 and N-2 as arearange .... if there is single one left, draw as line chart.

File: src/projection/util.ts [hunk 4]
```diff
@@ -13,47 +42,54 @@ export function toAreaRangeSeries(data: AreaRangeData, index: number): any[] {
...
+    const percentiledata97 = data.percentileData.find(
+        (_) => _.percentile == "97.5"
+    )?.data;
+
+    const output: any[] = [];
+        output.push({
+            name: `2.5% to 97.5%`,
```
Comment (lzl124631x) [inline src/projection/util.ts:64]: name should be generated from the percentile instead of hardcoding

File: src/projection/projection.tsx (hunk 2 above)
Comment (saipramod) [reply src/projection/projection.tsx:28]: oh shit i merged the backend just now. i will create another pr for this.

File: src/projection/projection.tsx (hunk 1 above)
Comment (saipramod) [reply src/projection/projection.tsx:5]: that makes sense.

File: src/projection/util.ts (hunk 3 above)
Comment (saipramod) [reply src/projection/util.ts:57]: i will handle this through another PR.

File: src/projection/util.ts (hunk 4 above)
Comment (saipramod) [reply src/projection/util.ts:64]: sure, i will handle this via another PR.

**Pull request merged**